src/btree/bt_read.c

```diff
@@ -152,6 +152,78 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
     return (ret);
 }
 
+/*
+ * __wt_modify_vector_init --
+ *     Initialise a modify vector.
+ */
+void
+__wt_modify_vector_init(WT_MODIFY_VECTOR *modifies, WT_SESSION_IMPL *session)
+{
+    WT_CLEAR(*modifies);
+    modifies->session = session;
+    modifies->listp = modifies->list;
+}
+
+/*
+ * __wt_modify_vector_push --
+ *     Push a modify update pointer to a modify vector. If we exceed the allowed stack space in the
+ *     vector, we'll be doing malloc here.
+ */
+int
+__wt_modify_vector_push(WT_MODIFY_VECTOR *modifies, WT_UPDATE *upd)
+{
+    WT_DECL_RET;
+    bool migrate_from_stack;
+
+    migrate_from_stack = false;
+
+    if (modifies->size >= WT_MODIFY_VECTOR_STACK_SIZE) {
+        if (modifies->capacity == 0 && modifies->size == WT_MODIFY_VECTOR_STACK_SIZE) {
+            migrate_from_stack = true;
+            modifies->listp = NULL;
+        }
+        WT_ERR(__wt_realloc_def(
+          modifies->session, &modifies->capacity, modifies->size + 1, &modifies->listp));
+        if (migrate_from_stack)
+            memcpy(modifies->listp, modifies->list, sizeof(modifies->list));
+    }
+    modifies->listp[modifies->size++] = upd;
+    return (0);
+
+err:
+    if (modifies->listp == NULL) {
+        modifies->listp = modifies->list;
+        modifies->capacity = 0;
+    }
+    return (ret);
+}
+
+/*
+ * __wt_modify_vector_pop --
+ *     Pop an update pointer off a modify vector.
+ */
+void
+__wt_modify_vector_pop(WT_MODIFY_VECTOR *modifies, WT_UPDATE **updp)
+{
+    WT_ASSERT(modifies->session, modifies->size > 0);
+
+    *updp = modifies->listp[--modifies->size];
+}
+
+/*
+ * __wt_modify_vector_free --
+ *     Free any resources associated with a modify vector. If we exceeded the allowed stack space on
+ *     the vector and had to fallback to dynamic allocations, we'll be doing a free here.
+ */
+void
+__wt_modify_vector_free(WT_MODIFY_VECTOR *modifies)
+{
+    if (modifies->capacity != 0)
+        __wt_free(modifies->session, modifies->listp);
+    WT_CLEAR(*modifies);
+    modifies->listp = modifies->list;
```

Comment:
Call `__wt_modify_vector_init()` instead of repeating the code?

Reply:
abcd390